doc/website/_docs/A01-sample-example.md

```diff
@@ -0,0 +1,205 @@
+---
+title: "ROHD Example"
+permalink: /docs/sample-example/
+excerpt: "Sample example of using ROHD."
+last_modified_at: 2022-12-05T08:48:05-04:00
+toc: true
+---
+
+Please make sure you had setup your ROHD Framework before running the test. Information of installation can be found at [setup page](/get-started/setup/).
+
+## A full example of a counter module
+
+To get a quick feel for what ROHD looks like, below is an example of what a simple counter module looks like in ROHD.
+
+```dart
+// Import the ROHD package
+import 'package:rohd/rohd.dart';
+
+// Define a class Counter that extends ROHD's abstract Module class
+class Counter extends Module {
+
+  // For convenience, map interesting outputs to short variable names for consumers of this module
+  Logic get val => output('val');
+
+  // This counter supports any width, determined at run-time
+  final int width;
+  Counter(Logic en, Logic reset, Logic clk, {this.width=8, String name='counter'}) : super(name: name) {
+    // Register inputs and outputs of the module in the constructor.
+    // Module logic must consume registered inputs and output to registered outputs.
+    en    = addInput('en', en);
+    reset = addInput('reset', reset);
+    clk   = addInput('clk', clk);
+
+    var val = addOutput('val', width: width);
+
+    // A local signal named 'nextVal'
+    var nextVal = Logic(name: 'nextVal', width: width);
+    
+    // Assignment statement of nextVal to be val+1 (<= is the assignment operator)
+    nextVal <= val + 1;
+
+    // `Sequential` is like SystemVerilog's always_ff, in this case trigger on the positive edge of clk
+    Sequential(clk, [
+      // `If` is a conditional if statement, like `if` in SystemVerilog always blocks
+      If(reset, then:[
+        // the '<' operator is a conditional assignment
+        val < 0
+      ], orElse: [If(en, then: [
+        val < nextVal
+      ])])
+    ]);
+  }
+}
+
+```
+
+<iframe src="https://zapp.run/embed/zkm06xqkn06?theme=dark&lazy=true&split=100" style="width: 100%; height: 20rem; border: 0; overflow: hidden;"></iframe>
```

Comment:
We discussed already, but we probably want to replace this with a link to the github codespace instead